extensions/parquet/table/src/main/java/io/deephaven/parquet/table/layout/ParquetFileHelper.java

```diff
@@ -5,14 +5,21 @@
 
 import io.deephaven.parquet.table.ParquetTableWriter;
 
-import java.nio.file.Path;
+import java.net.URI;
+import java.util.regex.Pattern;
 
 final class ParquetFileHelper {
     /**
      * Used as a filter to select relevant parquet files while reading all files in a directory.
      */
-    static boolean fileNameMatches(final Path path) {
-        final String fileName = path.getFileName().toString();
-        return fileName.endsWith(ParquetTableWriter.PARQUET_FILE_EXTENSION) && fileName.charAt(0) != '.';
+    private static final Pattern HIDDEN_FILE_PATTERN = Pattern.compile("(^|/)\\.[^/]+");
```

Comment:
What's the logic here? We'll skip a file if any directory or the file along the path starts with `.`? This seems pretty magic to me. Is this something that we should allow callers to configure?

Reply:
Yes, detecting any hidden parquet files or parquet files inside hidden subdirectories.
You are right, this does seems odd for S3. For files, it makes sense though. So I can keep the logic more file centered.